apps/builder/app/routes/rest.assets.tsx

```diff
@@ -0,0 +1,55 @@
+import type { ActionArgs, LoaderArgs } from "@remix-run/node";
+import { MaxAssets, type Asset } from "@webstudio-is/asset-uploader";
+import {
+  loadByProject,
+  initiateUpload,
+} from "@webstudio-is/asset-uploader/index.server";
+import { sentryException } from "~/shared/sentry";
+import { createContext } from "~/shared/context.server";
+import env from "~/env/env.server";
+
+export const loader = async ({
+  params,
+  request,
+}: LoaderArgs): Promise<Array<Asset>> => {
+  if (params.projectId === undefined) {
+    throw new Error("Project id undefined");
+  }
+  const context = await createContext(request);
+  return await loadByProject(params.projectId, context);
+};
+
+export const action = async (props: ActionArgs) => {
+  const { request } = props;
+
+  const context = await createContext(request);
+
+  try {
+    if (request.method === "POST") {
+      const formData = await request.formData();
+      const projectId = formData.get("projectId") as string;
+      const assetId = formData.get("assetId") as string;
+      const filename = formData.get("filename") as string;
+      if (projectId === null || assetId === null || filename === null) {
+        throw Error("Project id, asset id or filename are missing");
+      }
+      const name = await initiateUpload(
```

Comment:
I dislike the name as its not upload but name getter, 
something similar to `presigned url`, as only this name allows to upload file later.
`createPresignedFileName` or something better